yapapi/executor/__init__.py

```diff
@@ -138,16 +139,25 @@ async def submit(
         stack = self._stack
         emit = cast(Callable[[Event], None], self._wrapped_consumer.async_call)
 
-        # Creating allocation
-        if not self._budget_allocation:
-            self._budget_allocation = cast(
-                rest.payment.Allocation,
-                await stack.enter_async_context(
-                    self._payment_api.new_allocation(
-                        self._budget_amount, expires=self._expires + CFG_INVOICE_TIMEOUT
-                    )
-                ),
-            )
+        ids_to_decorate = []
+        if not self._budget_allocations:
+            async for account in self._payment_api.accounts():
+                allocation = cast(
+                    rest.payment.Allocation,
+                    await stack.enter_async_context(
+                        self._payment_api.new_allocation(
```

Comment:
It would be nice to use `account.address` as well when creating allocation.

Reply:
Thanks, I forgot about this.

Reply:
Fixed.